dev/tests/proxy.rs

```diff
@@ -395,19 +409,145 @@ async fn update_certificates() {
 
     proxy
         .db
-        .insert_proxy_route(&sr.backend_id, "foobar", &server.address.to_string())
+        .insert_proxy_route(&sr.backend_id, "foobar", &server.address.to_string(), None)
         .await
         .unwrap();
 
-    let result = proxy.http_get("foobar", "/").await.unwrap();
+    let result = proxy.http_get("foobar", "/", None, None).await.unwrap();
     assert_eq!("Hello World", result.text().await.unwrap());
 
     proxy.update_cert().unwrap();
     let new_cert = proxy.certs.cert_pem.clone();
 
-    let result = proxy.http_get("foobar", "/").await.unwrap();
+    let result = proxy.http_get("foobar", "/", None, None).await.unwrap();
     assert_eq!("Hello World", result.text().await.unwrap());
 
     // Ensure the certs are actually different.
     assert_ne!(original_cert, new_cert);
 }
+
+#[integration_test]
+async fn simple_missing_bearer_token() {
+    let proxy = Proxy::new(None).await.unwrap();
+    let server = Server::new(|_| async { "Hello World".into() })
+        .await
+        .unwrap();
+
+    let sr = base_spawn_request();
+    proxy.db.insert_backend(&sr).await.unwrap();
+    proxy
+        .db
+        .update_backend_state(&sr.backend_id, BackendState::Ready)
+        .await
+        .unwrap();
+
+    proxy
+        .db
+        .insert_proxy_route(&sr.backend_id, "foobar", &server.address.to_string(), Some("foobar"))
+        .await
+        .unwrap();
+
+    let result = proxy.http_get("foobar", "/", None, None).await.unwrap();
+    assert_eq!(StatusCode::UNAUTHORIZED, result.status());
+}
+
+#[integration_test]
+async fn simple_bearer_token() {
+    let proxy = Proxy::new(None).await.unwrap();
+    let server = Server::new(|_| async { "Hello World".into() })
+        .await
+        .unwrap();
+
+    let sr = base_spawn_request();
+    proxy.db.insert_backend(&sr).await.unwrap();
+    proxy
+        .db
+        .update_backend_state(&sr.backend_id, BackendState::Ready)
+        .await
+        .unwrap();
+
+    proxy
+        .db
+        .insert_proxy_route(&sr.backend_id, "foobar", &server.address.to_string(), Some("foobar"))
+        .await
+        .unwrap();
+
+    let result = proxy.http_get("foobar", "/", Some("foobar"), None).await.unwrap();
```

Comment:
Might be good to test the token-in-query-param use case, too.

Reply:
The `simple_cookie_set` test tests this endpoint, unless I'm misunderstanding?

Reply:
Oh right - somehow I missed that! Nevermind me!